build/common.sh

```diff
@@ -203,6 +207,18 @@ function kube::build::is_osx() {
   [[ "$(uname)" == "Darwin" ]]
 }
 
+function kube::build::update_dockerfile() {
+  if kube::build::is_osx; then
```

Comment:
You're only getting the first element by referencing $sed_opts.
Try ${sed_opts[@]}

Also try single quotes in the array element instead:
sed_opts=("-i ''")